src/nomad_parser_vasp/parsers/xml_parser.py

```diff
@@ -6,19 +6,19 @@
 )
 
 from nomad.config import config
-from nomad.parsing import MatchingParser
+from nomad.parsing import MatchingParserInterface
 from nomad.parsing.file_parser.mapping_parser import (
     MetainfoParser,
     XMLParser,
 )
-from schema_packages.vasp_package import Simulation
+from nomad_parser_vasp.schema_packages.vasp_package import Simulation
 
 configuration = config.get_plugin_entry_point(
     'nomad_parser_vasp.parsers:xml_entry_point'
 )
 
 
-class VasprunXMLParser(MatchingParser):
+class VasprunXMLParser(MatchingParserInterface):
```

Comment:
I suggest you do not inherit it here but in entry point load function you return MatchingParserInterface(lself.dict)

Reply:
Out of curiosity, why can `parser_class_name` not be passed as an argument too?
It seems like it should be set as a class variable.